### ISSUE TITLE
Private injection on smallrye-metrics

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#5071]
**Describe the bug**
Following this discussion on Zulip: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Unrecommended.20usage.20of.20private.20members/near/179447246 I spotted some private field injection on `smallrye-metrics`.

```
2019-10-31 10:36:03,769 WARN  [io.qua.arc.pro.BeanProcessor] (build-3) Found unrecommended usage of private members (use package-private instead) in framework beans:
	- Bean constructor void <init>(org.eclipse.microprofile.metrics.MetricRegistry registry) on io.smallrye.metrics.interceptors.MetricsInterceptor,
	- Bean constructor void <init>(javax.enterprise.inject.spi.Bean<?> bean, org.eclipse.microprofile.metrics.MetricRegistry registry) on io.smallrye.metrics.interceptors.TimedInterceptor,
	- Bean constructor void <init>(javax.enterprise.inject.spi.Bean<?> bean, org.eclipse.microprofile.metrics.MetricRegistry registry) on io.smallrye.metrics.interceptors.MeteredInterceptor,
	- Bean constructor void <init>(javax.enterprise.inject.spi.Bean<?> bean, org.eclipse.microprofile.metrics.MetricRegistry registry) on io.smallrye.metrics.interceptors.CountedInterceptor,
	- Bean constructor void <init>(javax.enterprise.inject.spi.Bean<?> bean, org.eclipse.microprofile.metrics.MetricRegistry registry) on io.smallrye.metrics.interceptors.ConcurrentGaugeInterceptor,
	- Producer method io.smallrye.metrics.interceptors.MetricNameFactory#metricName()
```

**Expected behavior**
No private injection for performance reason.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 0.27 or current master

**Comment by jmartisk** [comment]
@loicmathieu Sure thing. I will do the fix as well. https://github.com/smallrye/smallrye-metrics/issues/199

**Comment by jmartisk** [comment]
Hm, I realized these injections are in the Metrics TCK code as well. I'll see what can be done there, but AFAIU this performance problem is a bit Quarkus-specific and might not apply to general usage of CDI, so this could potentially hit some resistance. Or do we have any source saying that this should not be done in general? Anyway, I guess that as long as the TCKs pass, it's not a big deal for us.

**Comment by geoand** [comment]
Indeed the private member injection thing is Quarkus specific.

**Comment by jmartisk** [comment]
Okay, we'll leave TCKs as they are

**Comment by gsmet** [comment]
This will be fixed by upgrading SmallRye Metrics to 2.3.1.
